pkg/scale/uint128.go

```diff
@@ -161,3 +161,8 @@ func (u *Uint128) UnmarshalJSON(data []byte) error {
 	u.Lower = dec.Lower
 	return nil
 }
+
+// MarshalJSON converts Uint128 to []byte.
+func (u *Uint128) MarshalJSON() ([]byte, error) {
+	return []byte(u.String()), nil
```

Comment:
Being paranoid here, does it work for `(*Uint128)(nil)`

Reply:
nope.
I'll change to value receiver.